### PR TITLE
chore: migrate statement to tsx

### PR DESCRIPTION
**Pull request by aizad-deriv**
## Changes:

Please include a summary of the change and which issue is fixed below:
-   ...

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: packages/reports/src/Containers/statement.tsx [hunk 1]
```diff
@@ -53,7 +98,7 @@ const DetailsComponent = ({ message = '', action_type = '' }) => {
 };
 
 const getRowAction = row_obj => {
-    let action;
+    let action: any;
```
Comment (mahdiyeh-deriv) [inline packages/reports/src/Containers/statement.tsx]: change this any to the appropriate type

File: packages/reports/src/Containers/statement.tsx [hunk 2]
```diff
@@ -136,28 +181,31 @@ const Statement = ({
     const columns = getStatementTableColumnsTemplate(currency);
     const columns_map = columns.reduce((map, item) => {
         map[item.col_index] = item;
+        // console.log('map', map);
```
Comment (mahdiyeh-deriv) [inline packages/reports/src/Containers/statement.tsx]: remove this console

File: packages/reports/src/Components/market-symbol-icon-row.tsx [hunk 3]
```diff
@@ -4,14 +4,14 @@ import { Icon, Popover, IconTradeTypes } from '@deriv/components';
 import { getMarketName, getTradeTypeName } from '../Helpers/market-underlying';
 
 type TMarketSymbolIconRow = {
-    icon: string;
+    icon?: string | null;
```
Comment (akmal-deriv) [inline packages/reports/src/Components/market-symbol-icon-row.tsx]: A question: if the icon is an optional value. Are there any cases where the `null` passed and can we refactor those calls instead?

Comment (aizad-deriv) [reply packages/reports/src/Components/market-symbol-icon-row.tsx]: None seems that was a mistake I made from before. I have updated it in the new commit. 👍🏼

File: packages/reports/src/Containers/statement.tsx [hunk 4]
```diff
@@ -54,7 +54,7 @@ type TStatement = {
     date_from: number | null;
     date_to: number | null;
     error: string;
-    filtered_date_range: Record<string, any>;
+    filtered_date_range: Pick<TFilterComponent, 'filtered_date_range'>;
```
Comment (suisin-deriv) [inline packages/reports/src/Containers/statement.tsx]:
```suggestion
    filtered_date_range: React.ComponentProps<typeof FilterComponent>['filtered_date_range'];
```

File: packages/reports/src/Components/filter-component.tsx [hunk 5]
```diff
@@ -5,7 +5,7 @@ import { connect } from 'Stores/connect';
 import CompositeCalendar from './Form/CompositeCalendar';
 import { TRootStore } from 'Stores/index';
 
-type TFilterComponent = {
+export type TFilterComponent = {
```
Comment (suisin-deriv) [inline packages/reports/src/Components/filter-component.tsx]: Don't export the type if you only use it once. Try to use React.ComponentProps<typeof {ComponentName}>[{'value'}].

File: packages/reports/src/Containers/statement.tsx [hunk 6]
```diff
@@ -8,8 +8,8 @@ import { getSupportedContracts } from '_common/contract';
 import { connect } from 'Stores/connect';
 import { getStatementTableColumnsTemplate } from '../Constants/data-table-constants';
 import PlaceholderComponent from '../Components/placeholder-component';
-import AccountStatistics from '../Components/account-statistics';
-import FilterComponent from '../Components/filter-component';
+import AccountStatistics, { TAccountStatistics } from '../Components/account-statistics';
+import FilterComponent, { TFilterComponent } from '../Components/filter-component';
```
Comment (suisin-deriv) [inline packages/reports/src/Containers/statement.tsx]:
```suggestion
import FilterComponent from '../Components/filter-component';
```

**Review by akmal-deriv — APPROVED** [review]

File: packages/reports/src/Constants/data-table-constants.tsx [hunk 7]
```diff
@@ -279,12 +286,7 @@ export const getMultiplierOpenPositionsColumnsTemplate = ({
             if (is_footer) return localize('Total');
 
             return (
-                <MarketSymbolIconRow
-                    action={cell_value}
-                    key={row_obj.id}
-                    payload={row_obj.contract_info}
-                    should_show_multiplier={false}
-                />
+                <MarketSymbolIconRow key={row_obj.id} payload={row_obj.contract_info} should_show_multiplier={false} />
```
Comment (mahdiyeh-deriv) [inline packages/reports/src/Constants/data-table-constants.tsx:289]: in the `packages/reports/src/Components/market-symbol-icon-row.tsx` path the `show_description` should be optional in the `TMarketSymbolIconRow` type

Comment (aizad-deriv) [reply packages/reports/src/Constants/data-table-constants.tsx:289]: the  `show_description?: boolean;` in `TMarketSymbolIconRow` is already optional.

**Review by mahdiyeh-deriv — APPROVED** [review]

**Pull request merged**